scripts/src/java/org/oppia/android/scripts/xml/XmlSyntaxCheck.kt

```diff
@@ -0,0 +1,92 @@
+package org.oppia.android.scripts.xml
+
+import org.oppia.android.scripts.common.RepositoryFile
+import org.oppia.android.scripts.common.XML_SYNTAX_CHECK_FAILED_OUTPUT_INDICATOR
+import org.oppia.android.scripts.common.XML_SYNTAX_CHECK_PASSED_OUTPUT_INDICATOR
+import org.xml.sax.SAXParseException
+import java.io.File
+import javax.xml.parsers.DocumentBuilder
+import javax.xml.parsers.DocumentBuilderFactory
+
+/**
+ * Script for ensuring that all XML files in the repo are syntactically correct.
+ *
+ * Usage:
+ *   bazel run //scripts:xml_syntax_check -- <path_to_directory_root>
+ *
+ * Arguments:
+ * - path_to_directory_root: directory path to the root of the Oppia Android repository.
+ *
+ * Example:
+ *   bazel run //scripts:xml_syntax_check -- $(pwd)
+ */
+fun main(vararg args: String) {
+  // Path of the repo to be analyzed.
+  val repoPath = args[0] + "/"
+
+  // A list of all XML files in the repo to be analyzed.
+  val searchFiles = RepositoryFile.collectSearchFiles(
+    repoPath = repoPath,
+    expectedExtension = ".xml"
+  )
+
+  val allErrorsList = mutableListOf<Pair<SAXParseException, File>>()
+
+  // Builder factory which provides the builder to parse the XMl.
+  val builderFactory = DocumentBuilderFactory.newInstance()
+
+  searchFiles.forEach { file ->
+    val docBuilder = builderFactory.newDocumentBuilder()
+    val xmlSyntaxErrorHandler = XmlSyntaxErrorHandler()
+    docBuilder.setErrorHandler(xmlSyntaxErrorHandler)
+    parseXml(docBuilder, file)
+    val fileErrorList = xmlSyntaxErrorHandler.retrieveErrorList()
+    if (fileErrorList.isNotEmpty()) {
+      fileErrorList.forEach { error ->
+        allErrorsList.add(Pair(error, file))
```

Comment:
It'd be more Kotlin-idiomatic to compute the errors list as a result of this overall expression (similar to how we use fold/reduce/map/flatMap elsewhere). I suggest trying to rearrange this logic to compute the resulting list rather than mutating it within lambdas.

Reply:
Done.